src/main.rs

```diff
@@ -263,6 +270,10 @@ fn run() -> Result<(), Box<::std::error::Error>> {
         reader = Box::new(reader.take(length));
     }
 
+    if let Some(length) = matches.value_of("c").and_then(|n| n.parse::<u64>().ok()) {
```

Comment:
In order not to repeat the parsing and `Box::new` logic, you could use something like this:
``` rust
let length_arg = matches.value_of("length").or(matches.value_of("c"));

if let Some(length) = length_arg.and_then(...) {
    // ...
}
```

Reply:
Yea looks cleaner, will implement